xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/less/buttons.less

```diff
@@ -9,72 +9,79 @@
   #gradient > .vertical-three-colors(lighten(@color, 10%), @color, 50%, darken(@color, 5%));
 }
 
+.btn-background(@color) {
+  background-color: @color;
+}
+
 .btn-border(@color) {
   border: 1px solid @color;
   border-bottom-color: darken(@color, 10%);
-  .box-shadow(~"inset 0 1px 0 rgba(255,255,255,.2), 0 1px 2px rgba(0,0,0,.05)");
+}
+
+.btn-no-border(@color) {
+  border: 1px solid @color;
 }
 
 /* &-.button selectors are added in order to override the mapping and legacy usage of .button class */
 .btn {
   &-default {
-    .btn-gradient(lighten(@btn-default-bg, 5%));
+    .btn-background(lighten(@btn-default-bg, 5%));
     .btn-border(@btn-default-bg);
 
     &:hover, 
     &:focus {
-      .btn-gradient(@btn-default-bg);
+      .btn-background(@btn-default-bg);
       .btn-border(@btn-default-bg);
     }
   }
 
   &-primary {
-    .btn-gradient(@btn-primary-bg);
-    .btn-border(@btn-primary-bg);
+    .btn-background(@btn-primary-bg);
+    .btn-no-border(@btn-primary-bg);
 
     &:hover {
-      .btn-gradient(darken(@btn-primary-bg, 5%));
-      .btn-border(@btn-primary-bg);
+      .btn-background(darken(@btn-primary-bg, 5%));
+      .btn-no-border(darken(@btn-primary-bg, 5%));
     }
   }
 
   &-success, &-success.button {
-    .btn-gradient(@btn-success-bg);
+    .btn-background(@btn-success-bg);
     .btn-border(@btn-success-bg);
```

Comment:
Why doesn't this button follow the `btn-no-border` mixin like btn-primary?
I can see a small border on the bottom of all buttons except primary on hte screenshot after the PR.

(the same applies for btn-info, btn-warning and btn-danger at least)